### PR TITLE
front: switch from yarn to npm

### LINKED ISSUE #9143: front: switch from yarn to npm
### Description and goal

we use actually yarn, in order to integrate osrd-ui repo in osrd we need to switch to npm

### Acceptance criteria

- [ ] remove `yarn.lock` and `node-modules` (convert yarn.lock??)
- [ ] update `scripts` in `package.json`
- [ ] update yarn commands in `dockerfile`
- [ ] install libraries via `npm install` in order to retrieve `node-modules` & `package-lock.json`

### PR DESCRIPTION
**Pull request by Yohh**
close #9143

### CODE REVIEW

**Comment by emersion** [comment]
I had success running https://github.com/imsnif/synp

I think we'd be less likely to run into issues if we can use this tool instead of crossing fingers.

**Comment by emersion** [comment]
- Running `npm install` before renaming `resolutions` to `overrides` in `package.json` makes npm use `@rtk-query/oazapfts-patched` instead of our fork.
- Running `npm install` after renaming `resolutions` to `overrides` in `package.json` makes npm die with ".git can't be found" in `husky install`.
- Trying to upgrade `oazapfts` to the latest version won't work because it uses a monorepo and Git repo dependencies can't target a subdir.

**Review by shenriotpro — APPROVED** [review]
Thanks for the README update

File: front/docker/Dockerfile.devel
Comment (emersion) [inline front/docker/Dockerfile.devel]: I think `rm -rf node_modules` is no longer necessary now.

File: front/package.json
Comment (emersion) [inline front/package.json]: Hm, why did the version constraints in `package.json` change? I'd expect the `package.json` to stay as-is except for the `overrides` rename.

File: front/docker/Dockerfile.devel
Comment (Yohh) [reply front/docker/Dockerfile.devel]: yep, missed that, it's done

File: front/scripts/generate-types.sh
Comment (emersion) [inline front/scripts/generate-types.sh]: Nit: could use `npx` here instead to avoid having to use `--`. (But `npm exec` is fine as well.)

File: front/tsconfig.json
Comment (emersion) [inline front/tsconfig.json]: This exposes `node` specific types in the global scope... Is this necessary when switching to NPM?

File: front/package.json
Comment (Yohh) [reply front/package.json]: done

File: front/scripts/generate-types.sh
Comment (Yohh) [reply front/scripts/generate-types.sh]: nice, done

File: front/tsconfig.json
Comment (Yohh) [reply front/tsconfig.json]: it was about missing types while trying to migrate successfully, just forgot to remove it, it's done

**Comment by Khoyo** [comment]
```
[khoyo@odin osrd-hotfix]$ git grep yarn
front/README.md:- Install playwright dependencies `cd ./front/ && yarn playwright install --with-deps`
front/README.md:this script as you would to `yarn e2e-tests` or `yarn playwright test`.
front/docker/Dockerfile.playwright:COPY front/package.json front/yarn.lock /app/front/
front/docker/Dockerfile.playwright:RUN yarn install --frozen-lockfile
front/package-lock.json:        "yarn": "^1.7.0"
front/package-lock.json:        "yarn": ">=1"
front/package-lock.json:        "yarn": ">=1"
front/package-lock.json:        "yarn": "*"
front/package-lock.json:    "node_modules/yarn": {
front/package-lock.json:      "resolved": "https://registry.npmjs.org/yarn/-/yarn-1.22.22.tgz",
front/package-lock.json:        "yarn": "bin/yarn.js",
front/package-lock.json:        "yarnpkg": "bin/yarn.js"
scripts/run-front-playwright-container.sh:VERSION=$(yarn list --pattern playwright --json | jq -r '.data.trees[].name | split("@")[-1]' | sort -u)
scripts/run-front-playwright-container.sh:    osrd-playwright:latest yarn playwright test "${args[@]}"
```

The dependency in the lockfile is probably valid, the rest probably isn't.

**Comment by Yohh** [comment]
> ```
> [khoyo@odin osrd-hotfix]$ git grep yarn
> front/README.md:- Install playwright dependencies `cd ./front/ && yarn playwright install --with-deps`
> front/README.md:this script as you would to `yarn e2e-tests` or `yarn playwright test`.
> front/docker/Dockerfile.playwright:COPY front/package.json front/yarn.lock /app/front/
> front/docker/Dockerfile.playwright:RUN yarn install --frozen-lockfile
> front/package-lock.json:        "yarn": "^1.7.0"
> front/package-lock.json:        "yarn": ">=1"
> front/package-lock.json:        "yarn": ">=1"
> front/package-lock.json:        "yarn": "*"
> front/package-lock.json:    "node_modules/yarn": {
> front/package-lock.json:      "resolved": "https://registry.npmjs.org/yarn/-/yarn-1.22.22.tgz",
> front/package-lock.json:        "yarn": "bin/yarn.js",
> front/package-lock.json:        "yarnpkg": "bin/yarn.js"
> scripts/run-front-playwright-container.sh:VERSION=$(yarn list --pattern playwright --json | jq -r '.data.trees[].name | split("@")[-1]' | sort -u)
> scripts/run-front-playwright-container.sh:    osrd-playwright:latest yarn playwright test "${args[@]}"
> ```
> 
> The dependency in the lockfile is probably valid, the rest probably isn't.

you're right, I removed the useless mentions in `package-lock.json` and updated the playwright script, thank you

**Comment by emersion** [comment]
The `package-lock.json` references should be retained. In general this file shouldn't be edited by hand. It has "yarn" references because some of our dependencies depend on yarn.

File: front/package.json
Comment (emersion) [reply front/package.json]: Hm, unless I'm missing something, I still see these `package.json` changes?

Comment (Yohh) [reply front/package.json]: are you sure your branch is up to date? I don"t see it anymore...

edit: ok I missed some packages

Comment (Yohh) [reply front/package.json]: done

**Review by emersion — COMMENTED** [review]
The changes look good to me!

My only remaining worry is that during the various iterations of this PR, the `package-lock.json` file has evolved and contains unwanted changes. For instance, at some point some dependencies were upgraded in `package.json`. To ensure we keep the NPM dependency tree as similar as possible to yarn's (and resolve the `yarn.lock` conflicts with the dev branch), I'd suggest we re-generate `package-lock.json` from `yarn.lock` by running `npm install` on the dev branch and then using the resulting `package-lock.json` file as-is in this branch. Wdyt?

**Review by emersion — APPROVED** [review]
I've applied my suggestion above, rebased, and also had to add `--package-lock-only` for `npm list` to make it work without a prior `npm install`.

This LGTM, but it would be best if another maintainer could review as well!

**Review by ghost — APPROVED** [review]
Not tested but LGTM

**Comment by emersion** [comment]
Some platform-specific optional dependencies were missing from `package-lock.json` due to https://github.com/npm/cli/issues/4828, causing breakage on platforms other than Linux amd64. To fix it, dependencies with platform-specific sub-packages need to be re-installed to force npm to not look at `node_modules/`. I did the following:

```sh
npm install --save-dev rollup@4.23.0 @swc/core@1.7.26
git restore package.json
npm install
```

**Comment by Yohh** [comment]
everything works fine for me on macos and linux

**Comment by Yohh** [comment]
everything works fine for me on macos and linux

**Review by Math-R — APPROVED** [review]
Lgtm and tested !

**Review by clarani — APPROVED** [review]
LGTM :white_check_mark:

**Pull request merged**
